### PR TITLE
drivers: gicv3: GICR_TYPER_LAST mask width is 1 bit

### PR DESCRIPTION
**Pull request by valentinkorenblit**
Fix compiler warning: bitwise comparison always evaluates to false

### CODE REVIEW

File: drivers/interrupt_controller/intc_gicv3_priv.h [hunk 1]
```diff
@@ -67,7 +67,7 @@
 #define GICR_TYPER_AFFINITY_VALUE_MASK		0xFFFFFFFFUL
 #define GICR_TYPER_AFFINITY_VALUE_GET(_val)	MASK_GET(_val, GICR_TYPER_AFFINITY_VALUE)
 #define GICR_TYPER_LAST_SHIFT			4
-#define GICR_TYPER_LAST_MASK			0x10UL
+#define GICR_TYPER_LAST_MASK			0x1UL
```
Comment (carlocaione) [inline drivers/interrupt_controller/intc_gicv3_priv.h:70]: I see no usage of `GICR_TYPER_LAST_MASK`?

Comment (carlocaione) [reply drivers/interrupt_controller/intc_gicv3_priv.h:70]: Also, according to the documentation the `LAST` bit for `GICR_TYPER` is on bit `[4]` so it is correct to have the mask as `BIT(4) == 0x10UL`.

Comment (valentinkorenblit) [reply drivers/interrupt_controller/intc_gicv3_priv.h:70]: Hi Carlo,

Here: https://github.com/zephyrproject-rtos/zephyr/blob/b8f7c81dd3930ae1dde60dcf1daba32d03176816/drivers/interrupt_controller/intc_gicv3.c#L559

Comment (carlocaione) [reply drivers/interrupt_controller/intc_gicv3_priv.h:70]: oh, great, this file is basically subverting the usual meaning of the term `MASK`, even adding a new `MASK()` macro to switch between the canonical meaning and the special one, wonderful.

**Review by carlocaione — APPROVED** [review]

**Review by dkalowsk — APPROVED** [review]

**Comment by SgrrZhf** [comment]
Oops, this is a mistake introduced by me, thanks for correcting it. ^_^. @valentinkorenblit 
Perhaps I can refine this header file by using `GEN_MASK` and `FIELD_GET` for making it more readable later.

**Pull request merged**